File: frontend/miniapp/src/api/results.ts
import { ApiClient } from "./client";

export type PatientResultListItem = {
  result_id: string;
  title: string;
  date: string | null;
  status: string;
  has_pdf: boolean;
  lab_name: string | null;
  clinic_name: string | null;
  short_services_summary: string | null;
};

export type PatientResultDetails = {
  result_id: string;
  title: string;
  date: string | null;
  status: string;
  has_pdf: boolean;
  lab_name: string | null;
  clinic_name: string | null;
  services: string[];
  sections: Array<Record<string, unknown>>;
  indicators: Array<Record<string, unknown>>;
  pdf_open_url: string | null;
  pdf_download_url: string | null;
};

export class ResultsApi {
  constructor(private readonly client: ApiClient) {}

  list(): Promise<PatientResultListItem[]> {
    return this.client.get<PatientResultListItem[]>("/patient/results");
  }

  get(resultId: string): Promise<PatientResultDetails> {
    return this.client.get<PatientResultDetails>(`/patient/results/${encodeURIComponent(resultId)}`);
  }
}

File: src/presentation/patient_api/schemas/results.py
"""Pydantic-схемы patient-facing results API."""

from pydantic import BaseModel


class PatientLabResultListItemResponse(BaseModel):
    result_id: str
    title: str
    date: str | None
    status: str
    has_pdf: bool
    lab_name: str | None
    clinic_name: str | None
    short_services_summary: str | None


class PatientLabResultDetailsResponse(BaseModel):
    result_id: str
    title: str
    date: str | None
    status: str
    has_pdf: bool
    lab_name: str | None
    clinic_name: str | None
    services: list[str]
    sections: list[dict]
    indicators: list[dict]
    pdf_open_url: str | None
    pdf_download_url: str | None
